pennylane/interfaces/execution.py

```diff
@@ -618,74 +618,76 @@ def cost_fn(params, x):
-    #     raise ValueError("Gradient transforms cannot be used with mode='forward'")
+    #         if mode in ("forward", "best"):
+    #             # replace the forward execution function to return
+    #             # both results and gradients
+    #             execute_fn = set_shots(device, override_shots)(device.execute_and_gradients)
+    #             gradient_fn = None
+    #             _mode = "forward"
     #
-    # try:
-    #     mapped_interface = INTERFACE_MAP[interface]
-    # except KeyError as e:
-    #     raise ValueError(
-    #         f"Unknown interface {interface}. Supported " f"interfaces are {SUPPORTED_INTERFACES}"
-    #     ) from e
-    # try:
-    #     if mapped_interface == "autograd":
-    #         from .autograd import execute as _execute
-    #     elif mapped_interface == "tf":
-    #         import tensorflow as tf
+    #         elif mode == "backward":
+    #             # disable caching on the forward pass
+    #             execute_fn = qml.interfaces.cache_execute(batch_execute, cache=None)
     #
-    #         if not tf.executing_eagerly() or "autograph" in interface:
-    #             from .tensorflow_autograph import execute as _execute
-    #         else:
-    #             from .tensorflow import execute as _execute
-    #     elif mapped_interface == "torch":
-    #         from .torch import execute as _execute
-    #     else:  # is jax
-    #         _execute = _get_jax_execute_fn(interface, tapes)
-    # except ImportError as e:
-    #     raise qml.QuantumFunctionError(
-    #         f"{mapped_interface} not found. Please install the latest "
-    #         f"version of {mapped_interface} to enable the '{mapped_interface}' interface."
-    #     ) from e
+    #             # replace the backward gradient computation
+    #             gradient_fn = qml.interfaces.cache_execute(
+    #                 set_shots(device, override_shots)(device.gradients),
+    #                 cache,
+    #                 pass_kwargs=True,
+    #                 return_tuple=False,
+    #             )
     #
-    # res = _execute(
-    #     tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_diff=max_diff, mode=_mode
-    # )
+    #     elif mode == "forward":
+    #         # In "forward" mode, gradients are automatically handled
+    #         # within execute_and_gradients, so providing a gradient_fn
+    #         # in this case would have ambiguous behaviour.
+    #         raise ValueError("Gradient transforms cannot be used with mode='forward'")
+
+    try:
+        mapped_interface = INTERFACE_MAP[interface]
+    except KeyError as e:  # pragma: no cover #TODO: remove no cover once more support for new return types
+        raise ValueError(
+            f"Unknown interface {interface}. Supported " f"interfaces are {SUPPORTED_INTERFACES}"
+        ) from e
+    try:
+        if mapped_interface == "autograd":
+            from .autograd import execute as _execute

```

Comment:
if the mapped_interface is anything other than autograd, the below call to `_execute` is going to fail right?

Reply:
Reverted changes in this file for now.